spec/unit/foreman_smartproxy_rest_v3_spec.rb

```diff
@@ -22,6 +24,9 @@
   describe '#create' do
     it 'sends POST request' do
       provider.expects(:request).with(:post, 'api/v2/smart_proxies', {}, is_a(String)).returns(mock(:code => '201'))
```

Comment:
Either refactor the method to take the original data structure (hash) as the parameter and call .to_json on it internally, then you can use all of Mocha's parameter matchers on the original hash, or pass a block to .with() which runs JSON.load against the parameter and tests it.